### PR TITLE
Changes to make main.py dynamic

### PR DESCRIPTION
**Pull request by aatimlick**
The prompt_user in the original utils_main_prompts.py was hard-coded to a specific machine. A new script with a dynamic function has been created (utils_main_prompts_dynamic.py, prompt_user_dynamic) and is now being called in the main.py script. Other changes within the branch are experimental notebooks. The new scripts were tested and main.py now downloads the pkl and csv files automatically. There are function checks that should produce errors with more information if the path is not dynamic (prompt_user_dynamic_precheck). The naming system can be modified.

### CODE REVIEW

File: src/utils/utils_main_prompts_dynamic.py
Comment (Polichinel) [inline src/utils/utils_main_prompts_dynamic.py:1]: This seems super and it is obviously the right solution compared to the hard-coded one. This also means that at one point you should just remove the old one and only have this - but you can do that in another branch no rush

**Review by Polichinel — COMMENTED** [review]
Good - It looks super. As long as you make sure you're no commits behind main you can merge when you want

File: data/.gitignore
Comment (Polichinel) [inline data/.gitignore:1]: Btw, this single .gitignore should be needed as long as we have the root level .gitignore you can just delete them. But if you're having trouble with git not ignoring your csv's and pkl's let me know

Comment (aatimlick) [reply data/.gitignore:1]: Was having some problems with the .gitignore and some other issues of an https 400 error - might have fixed it by increasing the buffer size but was a temporary solution while troubleshooting.

**Pull request merged**